agent/structs/config_entry.go

```diff
@@ -297,3 +372,53 @@ type ServiceConfigResponse struct {
 
 	QueryMeta
 }
+
+// ConfigEntryResponse returns a single ConfigEntry
+type ConfigEntryResponse struct {
+	Entry ConfigEntry
+	QueryMeta
+}
+
+func (c *ConfigEntryResponse) MarshalBinary() (data []byte, err error) {
+	// bs will grow if needed but allocate enough to avoid reallocation in common
+	// case.
+	bs := make([]byte, 128)
+	enc := codec.NewEncoderBytes(&bs, msgpackHandle)
+
+	if err := enc.Encode(c.Entry.GetKind()); err != nil {
+		return nil, err
+	}
```

Comment:
Note to self: why do we need both map inspection of Kind and custom encoding with a type prefix here? Feels like we can use one approach or the other universally would be cleaner.